### DISCUSSION TITLE
Should I begin each page with a check for `useAuthenticationStatus` and `isLoading`?

### DISCUSSION THREAD

**Discussion by tyuen** [nhost/nhost#1078] [Q&A] [answered]
The help docs don't make it clear when I need to use `isLoading` which is returned by `useAuthenticationStatus`.

I only know I need to check for it on the Login page to see when the user is being logged in and after it has successfully logged in.

But if a user revisits a page after a few hours (assuming that the JWT has expired but the refresh token is still valid), should I expect `isAuthenticated=false` and `isLoading=true` or should I expect `isAuthenticated=true` for as long as the refresh token is still valid? In other words, should I also be detecting `isLoading` on every page?

When I use Apollo's `useQuery` with the `skip` property, can I just do `{skip: !userId}` or do I need to do `{skip: !userId || !isAuthenticated || isLoading}`? In other words, does `useQuery` automatically wait until the JWT is refreshed before continuing?

**Comment by elitan** [comment, marked as answer]
>  should I expect isAuthenticated=false and isLoading=true or should I expect isAuthenticated=true for as long as the refresh token is still valid?

When a user comes back after a few hours and has a refresh token, this will be the first state:

```
isLoading=true
isAuthenticated=false
```

The reason is, the client (user) must make a request to verify that the refresh token is valid.

> In other words, should I also be detecting isLoading on every page?

Yes

> When I use Apollo's useQuery with the skip property, can I just do {skip: !userId} or do I need to do {skip: !userId || !isAuthenticated || isLoading}? In other words, does useQuery automatically wait until the JWT is refreshed before continuing?

I think both of those would work so I would probably use the first one (`skip: !userId`) because it's less code.
